File: libatsc3Module/src/main/java/org/ngbp/libatsc3/ndk/a331/Service.java
package org.ngbp.libatsc3.ndk.a331;

import androidx.annotation.NonNull;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Locale;

public class Service {
    int serviceId;
    String globalServiceId;
    int majorChannelNo;
    int minorChannelNo;
    int serviceCategory;

    String shortServiceName;

    List<BroadcastSvcSignaling> broadcastSvcSignalingCollection = new ArrayList<>();

    public int getServiceId() {
        return serviceId;
    }

    public String getGlobalServiceId() {
        return globalServiceId;
    }

    public int getMajorChannelNo() {
        return majorChannelNo;
    }

    public int getMinorChannelNo() {
        return minorChannelNo;
    }

    public int getServiceCategory() {
        return serviceCategory;
    }

    public String getShortServiceName() {
        return shortServiceName;
    }

    public List<BroadcastSvcSignaling> getBroadcastSvcSignalingCollection() {
        return Collections.unmodifiableList(broadcastSvcSignalingCollection);
    }

    @NonNull
    public String toString() {
        return String.format(Locale.US, "%d.%d %s", majorChannelNo, minorChannelNo, shortServiceName);
    }
}

File: libatsc3Module/src/main/java/org/ngbp/libatsc3/entities/service/Service.kt
package org.ngbp.libatsc3.entities.service

import java.util.*
import kotlin.collections.ArrayList

data class Service(
        var serviceId: Int = 0,
        var globalServiceId: String? = null,
        var majorChannelNo: Int = 0,
        var minorChannelNo: Int = 0,
        var serviceCategory: Int = 0,
        var shortServiceName: String? = null,
        var broadcastSvcSignalingCollection: ArrayList<BroadcastSvcSignaling> = ArrayList()
) {
    override fun toString(): String {
        return String.format(Locale.US, "%d.%d %s", majorChannelNo, minorChannelNo, shortServiceName)
    }
}